AutoUsing/Analysis/Project.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Linq;
+using System;
+
+namespace AutoUsing
+{
+    public class Project : IDisposable
+    {
+        private XmlDocument Document { get; set; }
+        private XmlNamespaceManager NamespaceManager { get; set; }
+        private FileWatcher FileWatcher { get; set; }
+
+        public List<PackageReference> References { get; set; }
+        public string RootDirectory { get; private set; }
+        public string Name { get; private set; }
+        public string NuGetPackageRoot { get; private set; }
+        public string FilePath { get; private set; }
+        public string FileName { get; private set; }
+
+        public Project(string filePath, bool watch)
+        {
+            Document = new XmlDocument();
+            References = new List<PackageReference>();
+
+            // Namespace for msbuild.
+            NamespaceManager = new XmlNamespaceManager(Document.NameTable);
+            NamespaceManager.AddNamespace("x", "http://schemas.microsoft.com/developer/msbuild/2003");
+
+            // Essential info about given project file.
+            GetBasicInfo(filePath);
+
+            // Using NuGet Packages Location, We won't need to wait for project
+            // builds to get completions for referenced dependencies. As we'll query
+            // the dlls directly from the NuGet installation folder set by the user.
+            GetNuGetRootDirectory();
+
+            // Package References
+            GetPackageReferences();
+
+            // Optional: Watch for changes.
+            if (watch) Watch();
+        }
+
+        private void GetBasicInfo(string filePath)
```

Comment:
I feel that `Get` methods should always return a value. Please rename void `Get` methods to `Load` or something of a kind.